tsl/src/continuous_aggs/refresh.c

```diff
@@ -364,12 +354,33 @@ continuous_agg_refresh(PG_FUNCTION_ARGS)
 					(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
 					 (errmsg("relation \"%s\" is not a continuous aggregate", relname))));
 	}
+	return cagg;
+}
 
-	cagg_ht = ts_hypertable_get_by_id(cagg->data.mat_hypertable_id);
-	Assert(cagg_ht != NULL);
-	time_dim = hyperspace_get_open_dimension(cagg_ht->space, 0);
-	Assert(time_dim != NULL);
-	refresh_window.type = ts_dimension_get_partition_type(time_dim);
+static Oid
+cagg_get_partition_type(ContinuousAgg *cagg)
```

Comment:
```suggestion
cagg_get_partition_type(const ContinuousAgg *cagg)
```

unless calls below prevent it.